src/node.cc

```diff
@@ -4531,6 +4531,8 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
     } while (more == true);
   }
 
+  AsyncWrap::RunDestroyCbs(&env);
```

Comment:
Unfortunately that seems to change the timing of the `destroy` callback visibly and makes a couple of tests fail. I’ll try to get to it tomorrow.